payment-request/algorithms-manual.https.html

```diff
@@ -70,52 +70,33 @@
   promise_test(async t => {
     new PaymentRequest(methods, detailsNoShippingOptions, options);
     const request = new PaymentRequest(methods, details, options);
-    const events = [];
-    const p1 = new Promise(resolve => {
+    const handlerPromise = new Promise(resolve => {
       request[`on${eventName}`] = event => {
-        events.push(event);
-        resolve("handler");
+        // "prevent immediate propagation" flag is set.
+        // This listener below won't fire!
         event.updateWith(details);
+        resolve(event);
       };
     });
-    const p2 = new Promise(resolve => {
-      request.addEventListener(eventName, event => {
-        events.push(event);
-        resolve("listener-1");
-      });
+    // This listener should never fire because the
+    // the event handler caused "prevent immediate propagation" to be set.
+    request.addEventListener(eventName, async () => {
```

Comment:
Would it make sense to use `t.unreached_func("Second event listener should never fire")`?

Reply:
Oh, nice! I didn't know about `t.unreached_func`!